### PR TITLE
build: real time chat

### PR DESCRIPTION
**Pull request by AyaQunoo**

### CODE REVIEW

**Review by Basel-01 — APPROVED** [review]

File: client/src/components/ChatBox/index.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+const { Content, Sider } = Layout;
+import {
+  MessagesSiderHead,
+  ChooseChatButton,
+  User,
+  Message,
+  ChatUserHead,
+  SendMessageForm
+} from '..';
+import axios from 'axios';
```
Comment (devstm) [inline client/src/components/ChatBox/index.tsx:15]: axios is 3rd party module should be on the top

File: client/src/components/ChatBox/index.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  const [text, setText] = useState("");
+  const [onlineUsers, setOnlineUsers] = useState<UserObjectType[]>([]);
+  const [allMessages, setAllMessages] = useState<MessageObjectType[]>([]);
+  const [selectedUser, setSelectedUser] = useState<UserObjectType>({
+    userId: 0,
+    userName: '',
+  });
+  const chatWraper = useRef<HTMLDivElement>(null);
+
+  const makeArrayOfObjectsUnique = (arr: UserObjectType[]) => {
```
Comment (devstm) [inline client/src/components/ChatBox/index.tsx:31]: define this function in on onther file 
utils/services

File: client/src/components/ChatBox/index.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  useEffect(() => {
+    if (selectedUser.userId) {
+      const getAllMessages = async () => {
+        const { data } = await axios.get(`/api/v1/message/${selectedUser.userId}`)
+        let tempmsg = data.data.map(({ id, updatedAt, createdAt, ...rest }: any) => ({ ...rest, isOur: rest.sender === loggedInUserId }))
+        setAllMessages(tempmsg);
+      }
+      getAllMessages();
+    }
+  }, [selectedUser.userId]);
```
Comment (devstm) [inline client/src/components/ChatBox/index.tsx:83]: selectedUser not selectedUser.userId

File: client/src/components/ChatUserHead/index.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+          onClick={() => setCollapsed(!collapsed)}
+          style={{
+            fontSize: '16px',
+            width: 64,
+            height: 64,
+          }}
+        />
+      )}
+      <Image
+        src='https://th.bing.com/th/id/OIP.f3DM2upCo-p_NPRwBAwbKQHaHa?pid=ImgDet&rs=1'
```
Comment (devstm) [inline client/src/components/ChatUserHead/index.tsx:23]: use user image and this link should be in dalback api attrbute

**Review by devstm — CHANGES_REQUESTED** [review]

**Review by abdullah-alsharif — APPROVED** [review]

**Review by devstm — APPROVED** [review]

**Review by MohammedMahani — APPROVED** [review]

**Pull request merged**